pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -0,0 +1,269 @@
+    Set<String> existingDictColumns =
+        segmentReader.toSegmentDirectory().getColumnsWithIndex(ColumnIndexType.DICTIONARY);
+    Set<String> existingNoDictColumns = new HashSet<>();
+    for (String column : existingAllColumns) {
+      if (!existingDictColumns.contains(column)) {
+        existingNoDictColumns.add(column);
+      }
+    }
+
+    // From new column config.
+    Set<String> newNoDictColumns = _indexLoadingConfig.getNoDictionaryColumns();
+
+    for (String column : existingAllColumns) {
+      if (existingNoDictColumns.contains(column) && newNoDictColumns.contains(column)) {
+        // Both existing and new column is RAW forward index encoded. Check if compression needs to be changed.
+        if (shouldChangeCompressionType(column, segmentReader)) {
+          columnOperationMap.put(column, Operation.CHANGE_RAW_COMPRESSION_TYPE);
+        }
+      } else if (existingNoDictColumns.contains(column) && !newNoDictColumns.contains(column)) {
+        // TODO: Enable dictionary
+      } else if (!existingNoDictColumns.contains(column) && newNoDictColumns.contains(column)) {
+        // TODO: Disable dictionary.
+      } else {
+        // No changes necessary.
+      }
+    }
+
+    return columnOperationMap;
+  }
+
+  private boolean shouldChangeCompressionType(String column, SegmentDirectory.Reader segmentReader) {
+    ColumnMetadata existingColMetadata = _segmentMetadata.getColumnMetadataFor(column);
+
+    // TODO: Remove this MV column limitation.
+    if (!existingColMetadata.isSingleValue()) {
+      return false;
+    }
+
+    // The compression type for an existing segment can only be determined by reading the forward index header.
+    try {
+      ForwardIndexReader fwdIndexReader = LoaderUtils.getForwardIndexReader(segmentReader, existingColMetadata);
+      ChunkCompressionType existingCompressionType = fwdIndexReader.getCompressionType();
+      Preconditions.checkState(existingCompressionType != null,
+          "Existing compressionType cannot be null for raw forward index column=" + column);
+
+      // Get the new compression type.
+      ChunkCompressionType newCompressionType = null;
+      Map<String, ChunkCompressionType> newCompressionConfigs = _indexLoadingConfig.getCompressionConfigs();
+      if (newCompressionConfigs.containsKey(column)) {
+        newCompressionType = newCompressionConfigs.get(column);
+      }
+
+      // Note that default compression type (PASS_THROUGH for metric and LZ4 for dimension) is not considered if the
+      // compressionType is not explicitly provided in tableConfig. This is to avoid incorrectly rewriting the all
+      // forward indexes during segmentReload when the default compressionType changes.
+      if (newCompressionType == null || existingCompressionType == newCompressionType) {
+        return false;
+      }
+
+      return true;
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while changing compression for column: {}", column, e);
```

Comment:
nit: reword the message: "Caught exception while assessing whether the compression type has changed for column: "

Reply:
Refactored exception handling to throw Exception consistent with other indexes. Not needed anymore.